src/ast/nodes/shared/Node.ts

```diff
@@ -132,10 +132,11 @@ export class NodeBase extends ExpressionEntity implements ExpressionNode {
 	declare annotationPure?: boolean;
 	declare annotations?: RollupAnnotation[];
 
-	context: AstContext;
+	get context(): AstContext {
+		return this.scope.context;
```

Comment:
I think this is a good compromise. But I do not see why we do not directly access `this.scope.context` in all AST nodes and avoid the indirection.

Reply:
I can do that as an additional refactor. It saves a little bit of typing and was easier to get something working to test the memory impact without needing to touch as many files, but doesn't need to exist.